src/Common.OData.ApiExplorer/AspNet.OData/Routing/ODataRouteBuilder.cs

```diff
@@ -222,12 +222,14 @@ void AppendParametersFromConvention( StringBuilder builder, IEdmOperation operat
                 var actionParameters = Context.ParameterDescriptions.ToDictionary( p => p.Name, StringComparer.OrdinalIgnoreCase );
                 var parameter = parameters.Current;
                 var name = parameter.Name;
+                actionParameters.TryGetValue( name, out var mappedParameter );
```

Comment:
I'm still not sure I 100% grock the full use case, but I don't see anything glaring where this change is specific to your scenario. It appears to be a rational fallback, even if it shouldn't happen.

Let's simply change the entire setup to:

```c#
                var routeParameterName = name;
                
                if ( actionParameters.TryGetValue( name, out var mappedParameter ) )
                {
#if WEBAPI
                    routeParameterName = mappedParameter.ParameterDescriptor.ParameterName;
#elif API_EXPLORER
                    routeParameterName = mappedParameter.ParameterDescriptor.Name;
#else
                    routeParameterName = mappedParameter.Name;
#endif
                }
```